src/WebInterface.js

```diff
@@ -14,92 +15,113 @@ const Config = require('./Config');
   }
 
   init() {
+    util.log('[WEB] Initializing...');
+    this.generateSecret();
+    this.setUpGetRoutes();
 
-    const message = { message: "RanvierMUD API Endpoints: /npcs /players /items /rooms /help" };
-    // Routes for the API's GET response.
-    router.get('/',        (req, res) => res.json(message));
-    router.get('/npcs',    this.getResponseData(this.state.MobFactory, 'npcs'));
-    router.get('/players', this.getResponseData(this.state.PlayerManager, 'players'));     
-    router.get('/items',   this.getResponseData(this.state.ItemManager, 'items'));
-    router.get('/rooms',   this.getResponseData(this.state.RoomManager, 'rooms'));
-    router.get('/help',    this.getResponseData(this.state.HelpManager, 'helps'));
-
-    router.post('/authenticate', (req, res) => {
-      const accountName = req.body.name;
-      const account = this.state.AccountManager.findByName(accountName);
-      
-      if (!account) {
-        util.log(`[WEB] ${accountName} Auth Failed: Account not found.`);
-        return res.json({ success: false, message: `${accountName} was not found.`});
-      }
+    // Authentication route.
+    router.post('/authenticate', (req, res) => this.authenticate(req, res));
 
-      // Both passwords should be hashed... right?
-      if (account.password !== req.body.password) {
-        util.log(`[WEB] ${accountName} Auth Failed: Incorrect password.`);
-        return res.json({ success: false, message: `${accountName} password incorrect.`});
-      }
+    app.use('/api', router);
 
-      let secret;
-      try {
-        secret = fs.readFileSync(__dirname + '/../.ranvierSecret');
-      } catch(e) {
-        util.log(`[WEB] Authenticated requests not available without a .ranverSecret file...`);
-        return res.json({ success: false, message: 'Authentication failed, contact the administrator. '});
-      }
+    app.listen(this.port);
+    util.log(`[WEB]: Web API activated and running on port ${this.port}.`);
+    util.log(`[WEB]: Web API Secret: ${this.secret}`);
+  }
 
-      const token = jwt.sign(account, secret, {
-        expiresIn: "24h"
-      });
+  generateSecret() {
+    try {
+      this.secret = fs.readFileSync(this.secretFile);
+    } catch(e) {
+      const salt  = bcrypt.genSaltSync(10);
+      this.secret = bcrypt.hashSync('ranvier', salt);
+      fs.writeFileSync(this.secretFile, this.secret);
+    }
+  }
 
-      util.log(`[WEB] ${accountName} Auth Success!`);
-      return res.json({ success: true, token, message: `${accountName} authenticated.`});
-      
-    });
+  getSecret() {
+    return this.secret;
+  }
 
+  setUpGetRoutes() {
+    const message = { message: "RanvierMUD API Endpoints: /npcs /players /items /rooms /help" };
+    const { MobFactory, PlayerManager, ItemManager, RoomManager, HelpManager } = this.state;
 
-    app.use('/api', router);
+    // Routes for the API's GET response.
+    router.get('/',        (req, res) => res.json(message));
```

Comment:
This line here is pretty non-standard unless you're doing HATEOAS. This should either 404 because there's nothing to do or if you want to keep this message make the response code 405 to indicate that it's not supported